### ISSUE TITLE
Can the total_loss , iou/iou and iou/sec_iou somehow reflect the final result?

### ISSUE THREAD

**Issue by zhouweii234** [hkchengrex/STCN#58]
The `total_loss` is calculated by logits the network output and the cls_gt, `iou/iou` and `iou/sec_iou` is calculated by the mask the network output and the gt mask. So according to my understanding, `total_loss , iou/iou and iou/sec_iou` can somehow reflect the final result.

However, I modified your network and train it. The `total_loss` at iterations 149900 of s03 is `0.4624426`,  `iou/iou` is `0.9361632 `and `iou/sec_iou` is `0.9153692`. Since your raw network's `total_loss` at iterations 149900 of s03 is `0.4386`, `iou/iou` is `0.9358` and `iou/sec_iou` is `0.9276`. It seems like that my modified network will get a similar result with yours. But I get a very bad result at davis2016 and the `J&F-Mean` is `0.16`. The result of blackswan is also very strange.
![image](https://user-images.githubusercontent.com/62086490/133436930-04ffe1a0-bf89-47de-93cf-bc1f17f97e59.png)

Why did I get such a bad result when the `total_loss , iou/iou and iou/sec_iou` is close to yours? Does it because the `total_loss , iou/iou and iou/sec_iou` can not reflect the results ,or I do something wrong when I modified the eval network according to my modification to the training network?

**Comment by hkchengrex** [comment]
1. Yes, those metrics do somehow reflect the model's performance.
2. I suspect that there is an implementation problem somewhere. For example, the pipeline for evaluating 00001 is almost exactly the same as in training and it should perform similarly. One suspect is crop size -- if your model somehow overfitted to the crops (e.g. by learning abs. positional encoding) it might fail in evaluation with a different input size.

**Comment by zhouweii234** [comment]
Thank you for your reply! By the way, does the `abs. positional encoding` means `absolute positional encoding` ?

**Comment by hkchengrex** [comment]
Yes. It was just a guess though.

**Comment by hkchengrex** [comment]
Like, you can try to resize and crop in evaluation just to see if the same degradation happens.

**Comment by liuyvchi** [comment]
@zhouweii234 Hi. Do you find the reason for the bad results? I meet the same issue with you.
